controllers/topology/mergepatch.go

```diff
@@ -52,32 +53,92 @@ func newMergePatchHelper(original, modified client.Object, c client.Client) (*me
 		return nil, errors.Wrap(err, "failed to marshal modified object to json")
 	}
 
+	// Apply the modified object to the original one, merging the values of both;
+	// in case of conflicts, values from the modified object are preserved.
 	originalWithModifiedJSON, err := jsonpatch.MergePatch(originalJSON, modifiedJSON)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to apply modified json to original json")
 	}
 
-	data, err := jsonpatch.CreateMergePatch(originalJSON, originalWithModifiedJSON)
+	// Compute the merge patch that will align the original object to the target
+	// state defined above.
+	rawPatch, err := jsonpatch.CreateMergePatch(originalJSON, originalWithModifiedJSON)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create merge patch")
 	}
 
+	// Remove from the patch any change for metadata fields computed by the system or
+	// for the object status, which are not relevant for reconciling a topology.
+	patch, err := removePaths(rawPatch, [][]string{
+		{"metadata", "selfLink"},
+		{"metadata", "uid"},
+		{"metadata", "resourceVersion"},
+		{"metadata", "generation"},
+		{"metadata", "managedFields"},
+		{"status"},
+	})
```

Comment:
This might break over time, we should do an `allow list` instead to keep `spec` and only the metadata fields we want to compare.

As an example, some resources might also have other top level subresources (outside of spec and status); or k8s might add more into metadata which might not be supported

Reply:
@vincepri done, PTAL